internal/controller/nonadmindownloadrequest_controller.go

```diff
@@ -105,7 +159,15 @@ func (r *NonAdminDownloadRequestReconciler) SetupWithManager(mgr ctrl.Manager) e
 				}
 			},
 			// DeleteFunc: , TODO: if velero DownloadRequests gets cleaned up, delete this?
-		},
+		}, builder.WithPredicates(
```

Comment:
For other controllers we have predicate folder and handler folder to filter out on a relevant annotations and events, possibly here you also want to be similar ?

Check:
  -> Predicate: https://github.com/migtools/oadp-non-admin/blob/master/internal/controller/nonadminbackup_controller.go#L796-L798
  -> The above predicate is wrapped in the composite backup predicate: https://github.com/migtools/oadp-non-admin/blob/master/internal/predicate/compositebackup_predicate.go
  -> Implementation for the Velero predicate: https://github.com/migtools/oadp-non-admin/blob/master/internal/predicate/velerobackup_predicate.go
    -> Handler for the above event: https://github.com/migtools/oadp-non-admin/blob/master/internal/controller/nonadminbackup_controller.go#L801

Reply:
I'm trying to make it simpler I think composite is too complicated and only because you applied it via "global" event filter function rather than per Watch per For.

Reply:
good point reusing predicate

Reply:
not watching backup or restore tho so not applicable

Reply:
Sure, make it simpler as your controller is watching only one type, just an idea of getting files similarly stored. Don't make changes which will over complicate things, in NAB, NaBSL we are actually making decisions based on few things so was easier from one composite place.